R/derive_joined.R

```diff
@@ -352,6 +352,18 @@ derive_vars_joined <- function(dataset,
 
   if (is.null(new_vars)) {
     new_vars <- chr2vars(colnames(dataset_add))
+    preexisting_vars <- chr2vars(colnames(dataset))
+    preexisting_vars_no_by_vars <- preexisting_vars[which(!(preexisting_vars %in% by_vars))]
+    if (any(new_vars %in% preexisting_vars_no_by_vars)) {
+      err_msg <- sprintf(
+        paste(
+          "The following columns in `dataset_add` have naming conflicts with `dataset`,\n",
+          "please make the appropriate modifications to `new_vars`, with respect to:\n%s"
+        ),
+        enumerate(vars2chr(new_vars[which(new_vars %in% preexisting_vars_no_by_vars)]))
+      )
+      abort(err_msg)
+    }
```

Comment:
The check should also be performed if `new_vars` is not `NULL`. For example
```
> myd <- data.frame(day = c(1, 2, 3), val = c(0, 17, 21))
> derive_vars_joined(
+     myd,
+     dataset_add = myd,
+     new_vars = exprs(val),
+     order = exprs(day),
+     mode = "last",
+     filter_join = day < day.join
+ )
  day val val.join
1   1   0       21
2   2  17       21
3   3  21       NA
```
should produce an error instead of adding `val.join`.

Reply:
Removing the if NULL bracket just about broke every-other test/use case in the vignettes, so I'm not entirely sure we want to do this without making it drastically change our useability

Reply:
Sorry, I did not mean to remove the `if` condition but move the highlighted part out of the bracket:
```
  if (is.null(new_vars)) {
    new_vars <- chr2vars(colnames(dataset_add))
  }
  preexisting_vars <- chr2vars(colnames(dataset))
  preexisting_vars_no_by_vars <- preexisting_vars[which(!(preexisting_vars %in% by_vars))]
  duplicates <- intersect(replace_values_by_names(new_vars), preexisting_vars_no_by_vars)
  if (length(duplicates) > 0) {
    err_msg <- sprintf(
      paste(
        "The following columns in `dataset_add` have naming conflicts with `dataset`,\n",
        "please make the appropriate modifications to `new_vars`, with respect to:\n%s"
      ),
      enumerate(vars2chr(duplicates))
    )
    abort(err_msg)
  }
```
I tweaked the check code above to cover cases where variables in `new_vars` are renamed.

Reply:
Thanks! Adopted this into the newest commit